x-pack/protocol/src/main/java/org/elasticsearch/protocol/xpack/indexlifecycle/MoveToStepRequest.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.protocol.xpack.indexlifecycle;
+
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.support.master.AcknowledgedRequest;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class MoveToStepRequest extends AcknowledgedRequest<MoveToStepRequest> implements ToXContentObject {
+    static final ParseField CURRENT_KEY_FIELD = new ParseField("current_step");
+    static final ParseField NEXT_KEY_FIELD = new ParseField("next_step");
+    private static final ConstructingObjectParser<MoveToStepRequest, String> PARSER =
+        new ConstructingObjectParser<>("move_to_step_request", false,
+            (a, index) -> {
+                StepKey currentStepKey = (StepKey) a[0];
+                StepKey nextStepKey = (StepKey) a[1];
+                return new MoveToStepRequest(index, currentStepKey, nextStepKey);
+            });
+    static {
+        PARSER.declareObject(ConstructingObjectParser.constructorArg(), (p, name) -> StepKey.parse(p), CURRENT_KEY_FIELD);
+        PARSER.declareObject(ConstructingObjectParser.constructorArg(), (p, name) -> StepKey.parse(p), NEXT_KEY_FIELD);
+    }
+
+    private String index;
+    private StepKey currentStepKey;
+    private StepKey nextStepKey;
+
+    public MoveToStepRequest(String index, StepKey currentStepKey, StepKey nextStepKey) {
```

Comment:
I'm concerned that now that this is going to be a public-facing class, the implementation exposes too many details that we don't need for the user.

For example, a user would now have to know about `StepKey` when I guess they only want to move to a given step by name.

Why do we require the current step name? If you want to move to a step, it feels like you should only have to specify the step name that you want to move to?

Especially for `StepKey`, knowing the `phase` and `action` might be do-able, but I wouldn't expect a user to know the internal names for the steps that get generated from an action. What about making that optional for the request so a user only has to specify the `phase` and `action` (like "phase: hot, action: allocate")?

Reply:
+1 to this, there is always the low-level rest client for this, and we can revisit adding it at a later time if we change our minds.